src/main/java/org/example/Operations/CollectionOperations.java

```diff
@@ -0,0 +1,254 @@
+
+        HttpResponse<String> response = HttpRequestUtil.sendAndParseGET(formData, baseUrl, token, client.getClient());
+
+        return new Response(response.statusCode(), response.body());
+    }
+
+    public Response list(String token, String lpath) {
+        CollectionsListParams prop = new CollectionsListParams();
+        return this.list(token, lpath, prop);
+    }
+
+    // uses Permission enum for permission parameter
+    public Response set_permission(String token, String lpath, String entityName, Permission permission,
+                                   CollectionsSetPermissionParams prop) {
+        // contains parameters for the HTTP request
+        Map<Object, Object> formData = new HashMap<>();
+        formData.put("op", "set_permission");
+        formData.put("lpath", lpath);
+        formData.put("entity-name", entityName);
+        formData.put("permission", permission.getValue());
+        prop.getAdmin().ifPresent(val -> formData.put("admin", String.valueOf(val)));
+
+        HttpResponse<String> response = HttpRequestUtil.sendAndParsePOST(formData, baseUrl, token,
+                client.getClient());
+
+        return new Response(response.statusCode(), response.body());
+    }
+
+    public Response set_permission(String token, String lpath, String entityName, Permission permission) {
+        CollectionsSetPermissionParams prop = new CollectionsSetPermissionParams();
+        return this.set_permission(token, lpath, entityName, permission, prop);
+    }
+
+    public Response set_inheritance(String token, String lpath, int enable, CollectionsSetInheritanceParams prop) {
+        // contains parameters for the HTTP request
+        Map<Object, Object> formData = new HashMap<>();
+        formData.put("op", "set_inheritance");
+        formData.put("lpath", lpath);
+        formData.put("enable", String.valueOf(enable));
+        prop.getAdmin().ifPresent(val -> formData.put("admin", String.valueOf(val)));
+
+        HttpResponse<String> response = HttpRequestUtil.sendAndParsePOST(formData, baseUrl, token, client.getClient());
+
+        return new Response(response.statusCode(), response.body());
+    }
+
+    public Response set_inheritance(String token, String lpath, int enable) {
+        CollectionsSetInheritanceParams prop = new CollectionsSetInheritanceParams();
+        return this.set_inheritance(token, lpath, enable, prop);
+    }
+
+    public Response modify_permissions(String token, String lpath, List<ModifyPermissionsOperations> jsonParam,
+                                       CollectionsModifyPermissionParams prop) {
+        // Serialize the operations parameter to JSON
+        ObjectMapper mapper = new ObjectMapper();
+        String operationsJson = null;
+        try {
+            operationsJson = mapper.writeValueAsString(jsonParam);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
I was trying to deal with the unhandled exception, but I can look into ways of doing it differently?

Reply:
Sure, but I want to know why you feel catching the exception is necessary.

Let's say you allowed the exception to be thrown. What are the pros and cons of allowing that?

Reply:
Oh, I meant I didn't specifically have a reason for catching the exception vs adding the exception to method signature. I truthfully just let my IDE decide what to do with the unhandled exception. 
I guess by throwing the exception it may make it harder to see where the error came from.

Reply:
Did your IDE provide any reasoning behind the suggestion?

Reply:
No not really, it was just giving me options on what to do with the unhandled exception.

Reply:
Ok. Does your IDE force you to handle the `RuntimeException`s when called in the unit tests?

Reply:
No it does not.

Reply:
That must be because `RuntimeException` is an unchecked exception.

Ok. Let's remove the try-catch block and force the user to handle the exception.

Reply:
Ok, it's been removed.